meeting-notes/2019/Q1/2019-01-28--js-core-dev-team-weekly.md

```diff
@@ -0,0 +1,209 @@
+- Ask for general questions. Could be things like:
+  - I'm stuck with something, I don't know who to ask. Who knows who to ask?
+  - Who can help me with xyz?
+- Plan this week
+  - Select issues to work on
+- Review remaining issues if there is time left
+
+## Notes
+
+### Week Update
+
+@jacobheun
+- Done:
+  - Better custom lock example in datastore s3 [js-datastore-s3#13](https://github.com/ipfs/js-datastore-s3/pull/13)
+    - Also updated example for the latest js-ipfs
+  - Fix datastore s3 issue with multiple callbacks due to AWS lib [js-datastore-s3#14](https://github.com/ipfs/js-datastore-s3/pull/14)
+  - Now lead maintainer of js-multiaddr
+  - Created issue for tracking transport listening retry [js-libp2p#312](https://github.com/libp2p/js-libp2p/issues/312)
+  - Lots of reviewing
+- Blocked:
+- Next:
+  - Work to finalize [js-libp2p#312](https://github.com/libp2p/js-libp2p/issues/312) so work can start on that
+  - Finish libp2p daemon PR [js-libp2p-daemon#1](https://github.com/libp2p/js-libp2p-daemon/pull/1)
+  - Finalize rendezvous timeline
+
+@vmx
+- Done:
+  - new js-ipld API:
+    - API Docs: https://github.com/ipld/js-ipld/pull/185
+    - js-ipld itself: https://github.com/ipld/js-ipld/pull/190
+    - Changes on projects that depend on js-ipld:
+       - https://github.com/ipfs/js-ipfs-unixfs-importer/pull/20
+       - https://github.com/ipfs/js-ipfs-unixfs-exporter/pull/15
+       - https://github.com/ipfs/js-ipfs-mfs/pull/35
+       - https://github.com/ipfs/js-ipfs/pull/1848
+- Blocked:
+  - Not really blocked, but it would be cool to get the new IPLD API merged soon.
+- Next:
+  - Graphsync selectors, creating grammar and parser
+  - Make the new JS IPLD API use the JS IPLD Format API. Start with dag-pb and hope to get other formats picked up by others.
+
+
+@alanshaw
+- Done:
+    - Fixed up `_libp2pNode` to `libp2p` move [#1832](https://github.com/ipfs/js-ipfs/pull/1832)
+    - Fixed bundle `ipfs-http-client` in Meteor [#931](https://github.com/ipfs/js-ipfs-http-client/pull/931)
+    - Started migration to CID v1 base32 by default with PR to js-cid [#73](https://github.com/multiformats/js-cid/pull/73)
+    - Worked on upgrade to hapi 18 [#1844](https://github.com/ipfs/js-ipfs/pull/1844)
+        - `ipfs.add` response not buffered!
+        - net removal of nearly 800 LOC so far
+    - Fix to js.ipfs.io - still using old files API [#241](https://github.com/ipfs/js.ipfs.io/pull/241)
+    - Reviewed and merged PR adding support for `_dnslink` subdomain to `ipfs.dns` [#1843](https://github.com/ipfs/js-ipfs/pull/1843)
+- Blocked:
+    - `null`
+- Next:
+    - Finish off upgrade to hapi 18
+    - Continue CID v1 base32 default work ([step 2 here](https://github.com/ipfs/js-ipfs/issues/1440#issuecomment-407343797))
+    - Filecoin Friday!
+
+@vasco-santos
+
+Done:
+- DHT integration in `js-ipfs`
+  - CI is green now [ipfs/js-ipfs#856](https://github.com/ipfs/js-ipfs/pull/856)
```

Comment:
Has this been tested across NATs? We should set a PoWoW Zoom Call where we go through multiple experiments and try to break it

Reply:
Currently, the only NAT traversal that we have is through circuit relay. 

So, if we do a zoom and run local nodes, we should be able run `findPeer` successfully on each other id's. Then, we should be able to connect, at least over circuit relay. But yes, I agree that this would be a nice test to make!